docs/tm/udq_actionx/pyaction.tex

```diff
@@ -87,6 +87,56 @@ \section{The \pyaction{} keyword}
 
  
 \subsection{The different arguments}
+The \inlinecode{run()} function will be called with exactly five arguments which
+your implementation can use. The five arguments are:
+\begin{description}
+\item[\inlinecode{ecl\_state}:] An instance of the \inlinecode{Opm::EclipseState}
+  class - this is a representation of \emph{all static properties} in the model,
+  ranging from porosity to relperm tables. The content of the
+  \inlinecode{ecl\_state} is immutable - you are not allowed to change the static
+  properties at runtime\footnote{This could certainly be interesting, but this
+  is beyond the scope of the \pyaction{} keyword.}.
+\item[\inlinecode{schedule}:] An instance of the \inlinecode{Opm::Schedule}
+  class - this is a representation of all the content from the \kw{SCHEDULE}
+  section, notably all well and group information and the timestepping. Being
+  able to change the \kw{SCHEDULE} information runtime is certainly one of the
+  main motivations for this functionality, however due to the complexity of
+  the \inlinecode{Opm::Schedule} class (section \ref{schedule_design})
+  the recommended way to actually mutate the \inlinecode{Opm::Schedule} is
+  through the use of a dummy \actionx{} keyword (section
+  \ref{pyaction_actionx}).
+\item[\inlinecode{report\_step}:] This is an integer for the report step we
+  are currently working on. Observe that the \pyaction{} is called for every
+  simulator timestep, i.e. it will typically be called multiple time with
+  the same value for the $\mathrm{report\_step}$ argument.
```

Comment:
Seems like there is no way to determine whether we are at the end of a report step or am I missing something?

Reply:
The ACTIONX/PYACTION machinery is oriented around timesteps, the report step boundaries are not so relevant (the latter would have been simpler); but yes you are right: There is no way to determine whether we are at the end of the report step.